toolchain/driver/clang_runner.h

```diff
@@ -43,13 +45,36 @@ class ClangRunner : ToolRunnerBase {
   //
   // If `verbose` is passed as true, will enable verbose logging to the
   // `err_stream` both from the runner and Clang itself.
-  ClangRunner(const InstallPaths* install_paths, llvm::StringRef target,
+  ClangRunner(const InstallPaths* install_paths,
               llvm::IntrusiveRefCntPtr<llvm::vfs::FileSystem> fs,
               llvm::raw_ostream* vlog_stream = nullptr);
+  ~ClangRunner();
 
   // Run Clang with the provided arguments.
+  //
+  // This works to support all of the Clang commandline, including commands that
+  // use target-dependent resources like linking. When it detects such commands,
+  // it will build the necessary resources before running the command.
   auto Run(llvm::ArrayRef<llvm::StringRef> args) -> bool;
 
+  // Run Clang with the provided arguments and without any target-dependent
+  // resources.
+  //
+  // This method can be used to avoid building target-dependent resources when
+  // unnecessary, but not all Clang command lines will work correct.
+  // Specifically, compile-only commands will typically work, while linking will
+  // not.
+  auto RunTargetIndependentCommand(llvm::ArrayRef<llvm::StringRef> args)
+      -> bool;
+
+  // Builds the target-specific resource directory for Clang.
+  //
+  // There is a resource directory installed along side the Clang binary that
+  // contains all the target independent files such as headers. However, for
+  // target-specific files like runtimes, we build those on demand here and
+  // return the path.
+  auto BuildTargetResourceDir(llvm::StringRef target) -> std::string;
```

Comment:
Could this and `RunTargetIndependentCommand` be private, or are they called from elsewhere? If it's just for tests, maybe those could get friend access, so that the public API is limited?